app/src/main/java/org/greenstand/android/TreeTracker/signup/CredentialEntryView.kt

```diff
@@ -57,6 +57,10 @@ import org.greenstand.android.TreeTracker.view.DepthButton
 import org.greenstand.android.TreeTracker.view.LanguageButton
 import org.greenstand.android.TreeTracker.view.TopBarTitle
 import org.greenstand.android.TreeTracker.view.UserButton
+
+private lateinit var scaffoldState: SignUpState
+private lateinit var actionBarViewModel: SignupViewModel
```

Comment:
We shouldn't have global state like this. It makes it hard to debug/reason about the code and will most likely introduce bugs now or later.

Reply:
We need to find a different way to fix this. This may work in the short run, but it's basically asking for complicated bugs in the future. I also think there are probably edge cases here that have not been tested. With a global viewmodel, one signup input can persist when going back to the screen.